lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8265,6 +8266,19 @@ ClangImporter::Implementation::importSwiftAttrAttributes(Decl *MappedDecl) {
   };
   importAttrsFromDecl(ClangDecl);
 
+  // Unannotated C++ types are considered unsafe.
+  // TODO: add logic for safe cases like aggregates of escapable types.
+  if (const auto *record = dyn_cast<clang::CXXRecordDecl>(ClangDecl)) {
+    if (SwiftContext.LangOpts.SafeCXXInterop &&
+        SwiftContext.LangOpts.hasFeature(Feature::AllowUnsafeAttribute) &&
+        !importer::hasNonEscapableAttr(record) &&
+        !importer::hasEscapableAttr(record) &&
+        !MappedDecl->getAttrs().hasAttribute<UnsafeAttr>()) {
```

Comment:
Please pull these checks for "should this type be imported as @unsafe?" into a separate function where we can hone the logic. There's already one place in the Clang importer where we add the `UnsafeAttr` (in response to an `@unsafe` SwiftAttr): let's consolidate this with that. 

I also think there are more things we should check for: types that are imported as foreign reference types (e.g., CF types, types with SWIFT_SHARED_REFERENCE or the immortal counterpart) should also be considered "safe". I should hope there are also structural rules we should consider in the future.

Reply:
I want to keep each individual PR as small as possible, I definitely think this logic will get more sophisticated over time. That being said, I think CF and shared reference types are easy to add here and now and will make a big impact. I plan to address the structural rules in a follow-up PR.

Reply:
Actually, it looks like `SWIFT_IMMORTAL_REFERENCE` and `SWIFT_UNSAFE_REFERENCE` currently expands to the exact same set of macros, so I'd like to defer dealing with the different foreign reference types  to a follow-up PR. But Added the exception for the CF types.